nucliadb_vectors2/src/vectors/data_point_provider/mod.rs

```diff
@@ -73,32 +73,22 @@ impl Index {
         }
         Ok(())
     }
-    fn new(at: &Path) -> VectorR<(Index, SLock)> {
+    pub fn new(at: &Path, do_sanity_check: bool) -> VectorR<Index> {
```

Comment:
Can you create an enumeration instead of using that `do_sanity_check` please ? It will improve code readability (at call site) and encode the meaning of that boolean by using the type system.

Reply:
Actually I prefer for it to be a boolean, since in my opinion it expresses better the choice the caller has: do a sanity check on creation or not.
Also, if I were to create an enum only for this option the caller would need  to import such enum. I think is too tedious for expressing what in essence is a boolean flag.

Reply:
There are plenty of blog posts that explain better than me why using boolean (and boolean flag) as parameter is a code smell.
Even if the user choice is just do or do not perform a sanity check, using a boolean does not express that choice actually and close definitely any improvement from that two choices to more options.

```rust
let index = Index::new("my_path", true); // Or Index::new("my_path", false)
```
vs
```rust
let index = Index::new("my_path", IndexCheck::Sanity); // Or Index::new("my_path", IndexCheck::Skip)
```

Reply:
Or, instead of introducing an enumeration, you can also create two separate constructors :
- `new` that just creates the index
- `with_sanity_check` that creates the index and perform the sanity check

Reply:
will do